Source/WebCore/workers/WorkerMessagingProxy.cpp

```diff
@@ -124,14 +124,20 @@ void WorkerMessagingProxy::startWorkerGlobalScope(const URL& scriptURL, PAL::Ses
     IDBClient::IDBConnectionProxy* proxy = document.idbConnectionProxy();
 
     SocketProvider* socketProvider = document.socketProvider();
+    
+#if ENABLE(NOISY_APIS)
+    IdentifierProvider* identifierProvider = document.identifierProvider();
```

Comment:
You could just have document.identifierProvider return null if ENABLE(NOISY_APIS) is 0 instead of having a check here.